torchrec/distributed/sharding/rw_sharding.py

```diff
@@ -288,7 +288,7 @@ def __init__(
             ),
         )
         self._dist = KJTAllToAll(
-            pg=pg,
+            pg=dist.new_group(),
```

Comment:
It's not recommended to create a separate Process Group here; and could results in performance regressions.

Reply:
Indeed, when there are too many tables, there will be a lot of overhead in stream synchronization.